### PR TITLE
JBPM-9979 Mark async active nodes and change default active node border color

### PR DESCRIPTION
**Pull request by bxf12315**
…and re-execute it when resuming process flow

**Thank you for submitting this pull request**

**JIRA**: _(please edit the JIRA link if it exists)_ 

[https://issues.redhat.com/browse/JBPM-9979](JBPM-9979)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* paste the link(s) from GitHub here
* link 2
* link 3 etc.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins retest this

**Comment by bxf12315** [comment]
Jenkins retest this

**Comment by nmirasch** [comment]
@bxf12315  this code has not relationship with the jira linked "Being able to park process execution on failed script task and re-execute it when resuming process flow", If we need to add the error color painting we need to add another jira explaining the feature even that it can be linked with this  jira if that makes sense.

**Comment by bxf12315** [comment]
> @bxf12315 this code has not relationship with the jira linked "Being able to park process execution on failed script task and re-execute it when resuming process flow", If we need to add the error color painting we need to add another jira explaining the feature even that it can be linked with this jira if that makes sense.
updated

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ProcessServiceIntegrationTest.java [hunk 1]
```diff
@@ -473,7 +473,8 @@ public void testProcessVariablesConversionDataTypes() {
             
             short age = 44;
             processClient.setProcessVariable(CONTAINER_ID, processInstanceId, "age", Short.toString(age) );
-            assertEquals( age, ((Number)processClient.getProcessInstanceVariable(CONTAINER_ID, processInstanceId, "age")).shortValue());
+            Short.parseShort(processClient.getProcessInstanceVariable(CONTAINER_ID, processInstanceId, "age").toString());
```
Comment (nmirasch) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ProcessServiceIntegrationTest.java:476]: @bxf12315 this change is unrelated with the error node border painting

**Comment by bxf12315** [comment]
@nmirasch  updated.

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json [hunk 2]
```diff
@@ -22,11 +22,42 @@
...
+          "annotation": "@javax.ws.rs.DefaultValue(\"#1e90ff\")",
+          "attribute": "value",
+          "oldValue": "\"#FF0000\"",
+          "newValue": "\"#1e90ff\"",
+          "package": "org.kie.server.remote.rest.jbpm.ui",
+          "classSimpleName": "ImageResource",
+          "methodName": "getProcessInstanceImage",
+          "parameterIndex": "5",
+          "elementKind": "parameter",
+          "justification": "[JBPM-9945] [GSS-RFE] [Runtime View] Being able to park process execution on failed script task and re-execute it when resuming process flow"
```
Comment (nmirasch) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json:43]: @bxf12315 There are changes that shouldn't be part of this, in example this justification is out of date. This file needs to be regenerated from scratch

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ProcessServiceIntegrationTest.java [hunk 3]
```diff
@@ -474,7 +474,7 @@ public void testProcessVariablesConversionDataTypes() {
             short age = 44;
             processClient.setProcessVariable(CONTAINER_ID, processInstanceId, "age", Short.toString(age) );
             assertEquals( age, ((Number)processClient.getProcessInstanceVariable(CONTAINER_ID, processInstanceId, "age")).shortValue());
-            
+
```
Comment (nmirasch) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ProcessServiceIntegrationTest.java:477]: @bxf12315  ?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/test/java/org/kie/server/services/jbpm/ui/ImageServiceBaseTest.java [hunk 4]
```diff
@@ -201,6 +201,112 @@ public void testLoopSubProcess() throws Exception {
...
+    @Test
+    public void testActiveNodeBorderColor() throws Exception {
+        String containerId = "test-container";
+        String processId = "test-processId";
+        long processInstanceId = 10;
+        String humanTaskOne = "_C8E8A0C7-ECC6-4C28-B1FE-D6DE5999239E";
+        String oneScriptNode = "_BA99E908-87C6-46C9-99ED-901DFCD1A2AA";
+        String twoScriptNode = "_1C98770B-2C62-41F1-AFBF-7B138EE5270E";
+        String threeScriptNode = "_D7C7D018-DC2D-4B97-BEFF-55B50F201103";
+        byte[] byteArray = getInputStreamAsByteArray(ImageServiceBaseTest.class.getResourceAsStream("/BAPL-2159-svg.svg"));
```
Comment (nmirasch) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/test/java/org/kie/server/services/jbpm/ui/ImageServiceBaseTest.java:266]: @bxf12315 would you mind to rename the svg to not to be linked with that BAPL I don't think we need to keep the reference to that BAPL this svg regarding it's just to try the error and active color painting test.

**Comment by bxf12315** [comment]
@nmirasch updated

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/test/java/org/kie/server/services/jbpm/ui/ImageServiceBaseTest.java [hunk 5]
```diff
@@ -201,6 +199,112 @@ public void testLoopSubProcess() throws Exception {
...
+        NodeInstanceDesc nodeInstanceDescActive = new org.jbpm.kie.services.impl.model.NodeInstanceDesc("1", humanTaskOne, "", "HumanTask", "", 1L, null,
+                                                                                                        "", 0, 1L, 3L, "", null, 1);
+
+        NodeInstanceDesc nodeInstanceDescCompleted2 = new org.jbpm.kie.services.impl.model.NodeInstanceDesc("2", oneScriptNode, "", "ScriptNode", "", 1L, null,
+                                                                                                            "", 0, 1L, 1L, "", null, 1);
+
+        NodeInstanceDesc nodeInstanceDescCompleted3 = new org.jbpm.kie.services.impl.model.NodeInstanceDesc("3", twoScriptNode, "", "ScriptNode", "", 1L, null,
+                                                                                                            "", 0, 1L, 4L, "", null, 1);
+
+        NodeInstanceDesc nodeInstanceDescError = new org.jbpm.kie.services.impl.model.NodeInstanceDesc("4", threeScriptNode, "", "ScriptNode", "", 1L, null,
```
Comment (nmirasch) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/test/java/org/kie/server/services/jbpm/ui/ImageServiceBaseTest.java:280]: @bxf12315 is confusing call this NodeInstanceDesc where is not in error state, could you rename it?

**Review by nmirasch — COMMENTED** [review]
Added some comments about tests

**Comment by bxf12315** [comment]
Done

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json [hunk 6]
```diff
@@ -22,11 +22,42 @@
...
     }
   },
-
   "ignores": {
     "revapi": {
-    "_comment": "Changes between 7.48.0.Final and the current branch. These changes are desired and thus ignored.",
-    "ignore": [
+      "_comment": "Changes between 7.48.0.Final and the current branch. These changes are desired and thus ignored.",
+      "ignore": [
+        {
```
Comment (nmirasch) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json:29]: @bxf12315 the two first blocks at "ignore" section  have to be removed, the first good one is the one that makes reference to [JBPM-9740] in "justification".

Comment (bxf12315) [reply kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json:29]: the two sections is rest api change and swagger change's comment.

File: jbpm-process-svg/src/main/java/org/jbpm/process/svg/SVGImageProcessor.java [hunk 7]
```diff
@@ -92,6 +94,12 @@ public static String transform(InputStream svg, List<String> completed, List<Str
     public static String transform(InputStream svg, List<String> completed, List<String> active,
                                    Map<String, String> subProcessLinks, String completedNodeColor,
                                    String completedNodeBorderColor, String activeNodeBorderColor, Map<String, Long> badges) {
+        return transform(svg, completed, active, Collections.emptyList(), subProcessLinks, completedNodeColor, completedNodeBorderColor, activeNodeBorderColor, badges, "");
+    }
+
+    public static String transform(InputStream svg, List<String> completed, List<String> active, List<String> error,
```
Comment (jstastny-cz) [inline jbpm-process-svg/src/main/java/org/jbpm/process/svg/SVGImageProcessor.java:100]: The updated method signature is not perfect. I'd suggest moving errorNodeBorderColor near other node/border color arguments. Plus, does an additional method make sense? I would say we should change signature of an existing method, cause as long as we provide any border/node color argument we might want to provide the newly introduced `errorNodeBorderColor` too.

File: jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/JBPMDesignerSVGProcessor.java [hunk 8]
```diff
@@ -86,4 +86,9 @@ public void processNodes(NodeList nodes) {
             processNodes(node.getChildNodes());
         }
     }
+
+    @Override
```
Comment (jstastny-cz) [inline jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/JBPMDesignerSVGProcessor.java:90]: Can we move this closer to other default*Transformation methods?

File: jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/SVGProcessor.java [hunk 9]
```diff
@@ -25,4 +26,6 @@
     String getSVG();
 
     void processNodes(NodeList nodes);
+
+    void defaultErrorTransformation(String nodeId, String errorNodeBorderColor);
```
Comment (jstastny-cz) [inline jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/SVGProcessor.java:30]: Here too, can we move closer to default*Transformation methods?

File: jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/StunnerSVGProcessor.java [hunk 10]
```diff
@@ -180,4 +180,15 @@ public void processNodes(NodeList nodes) {
             processNodes(node.getChildNodes());
         }
     }
+
```
Comment (jstastny-cz) [inline jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/StunnerSVGProcessor.java:183]: Same here, can we move closer to other default*Transformation methods?

File: kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/impl/UIServicesClientImplTest.java [hunk 11]
```diff
@@ -29,35 +29,34 @@ public void testCreateColorUrlParams() {
...
 
         assertEquals(RestURI.SVG_NODE_COMPLETED_COLOR + "=" + uiServicesClientImpl.encode(completedNodeColor) +
                              "&" + RestURI.SVG_NODE_COMPLETED_BORDER_COLOR + "=" + uiServicesClientImpl.encode(completedNodeBorderColor) +
                              "&" + RestURI.SVG_NODE_ACTIVE_COLOR + "=" + uiServicesClientImpl.encode(activeNodeBorderColor),
-                     uiServicesClientImpl.createColorURLParams(completedNodeColor, completedNodeBorderColor, activeNodeBorderColor));
+                     uiServicesClientImpl.createColorURLParams(completedNodeColor, completedNodeBorderColor, activeNodeBorderColor, ""));
 
         assertEquals(RestURI.SVG_NODE_COMPLETED_COLOR + "=" + uiServicesClientImpl.encode(completedNodeColor),
-                     uiServicesClientImpl.createColorURLParams(completedNodeColor, "", ""));
+                     uiServicesClientImpl.createColorURLParams(completedNodeColor, "", "",""));
```
Comment (jstastny-cz) [inline kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/impl/UIServicesClientImplTest.java:46]:
```suggestion
                     uiServicesClientImpl.createColorURLParams(completedNodeColor, "", "", ""));
```

File: kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/impl/UIServicesClientImplTest.java [hunk 12]
```diff
@@ -29,35 +29,34 @@ public void testCreateColorUrlParams() {
...
         assertEquals(RestURI.SVG_NODE_COMPLETED_BORDER_COLOR + "=" + uiServicesClientImpl.encode(completedNodeBorderColor),
-                     uiServicesClientImpl.createColorURLParams("", completedNodeBorderColor, ""));
+                     uiServicesClientImpl.createColorURLParams("", completedNodeBorderColor, "", ""));
         assertEquals(RestURI.SVG_NODE_ACTIVE_COLOR + "=" + uiServicesClientImpl.encode(activeNodeBorderColor),
-                     uiServicesClientImpl.createColorURLParams("", "", activeNodeBorderColor));
+                     uiServicesClientImpl.createColorURLParams("", "", activeNodeBorderColor,""));
```
Comment (jstastny-cz) [inline kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/impl/UIServicesClientImplTest.java:50]:
```suggestion
                     uiServicesClientImpl.createColorURLParams("", "", activeNodeBorderColor, ""));
```

File: kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/impl/UIServicesClientImplTest.java [hunk 13]
```diff
@@ -29,35 +29,34 @@ public void testCreateColorUrlParams() {
...
 
         assertEquals(RestURI.SVG_NODE_COMPLETED_COLOR + "=" + uiServicesClientImpl.encode(completedNodeColor) +
                              "&" + RestURI.SVG_NODE_COMPLETED_BORDER_COLOR + "=" + uiServicesClientImpl.encode(completedNodeBorderColor),
-                     uiServicesClientImpl.createColorURLParams(completedNodeColor, completedNodeBorderColor, ""));
+                     uiServicesClientImpl.createColorURLParams(completedNodeColor, completedNodeBorderColor, "",""));
```
Comment (jstastny-cz) [inline kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/impl/UIServicesClientImplTest.java:54]:
```suggestion
                     uiServicesClientImpl.createColorURLParams(completedNodeColor, completedNodeBorderColor, "", ""));
```

File: kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/impl/UIServicesClientImplTest.java [hunk 14]
```diff
@@ -29,35 +29,34 @@ public void testCreateColorUrlParams() {
...
         assertEquals(RestURI.SVG_NODE_COMPLETED_BORDER_COLOR + "=" + uiServicesClientImpl.encode(completedNodeBorderColor) +
                              "&" + RestURI.SVG_NODE_ACTIVE_COLOR + "=" + uiServicesClientImpl.encode(activeNodeBorderColor),
-                     uiServicesClientImpl.createColorURLParams("", completedNodeBorderColor, activeNodeBorderColor));
+                     uiServicesClientImpl.createColorURLParams("", completedNodeBorderColor, activeNodeBorderColor,""));
```
Comment (jstastny-cz) [inline kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/impl/UIServicesClientImplTest.java:57]:
```suggestion
                     uiServicesClientImpl.createColorURLParams("", completedNodeBorderColor, activeNodeBorderColor, ""));
```

File: kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/impl/UIServicesClientImplTest.java [hunk 15]
```diff
@@ -29,35 +29,34 @@ public void testCreateColorUrlParams() {
...
         assertEquals(RestURI.SVG_NODE_COMPLETED_COLOR + "=" + uiServicesClientImpl.encode(completedNodeColor) +
                              "&" + RestURI.SVG_NODE_ACTIVE_COLOR + "=" + uiServicesClientImpl.encode(activeNodeBorderColor),
-                     uiServicesClientImpl.createColorURLParams(completedNodeColor, "", activeNodeBorderColor));
+                     uiServicesClientImpl.createColorURLParams(completedNodeColor, "", activeNodeBorderColor,""));
```
Comment (jstastny-cz) [inline kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/impl/UIServicesClientImplTest.java:60]:
```suggestion
                     uiServicesClientImpl.createColorURLParams(completedNodeColor, "", activeNodeBorderColor, ""));
```

File: kie-server-parent/kie-server-remote/kie-server-client/src/build/revapi-config.json [hunk 16]
```diff
@@ -1,57 +1,67 @@
...
-                            "org.kie.server.client.helper.JBPMServicesClientBuilder",
-                            "org.kie.server.client.helper.JBPMUIServicesClientBuilder",
-                            "org.kie.server.client.helper.OptaplannerServicesClientBuilder",
-                            "org.kie.server.client.KieServicesException",
-                            "org.kie.server.client.KieServicesFactory"
-                        ]
-                    }
-                }
-            }
+  "filters": {
```
Comment (jstastny-cz) [inline kie-server-parent/kie-server-remote/kie-server-client/src/build/revapi-config.json:2]: Can we keep original formatting of the json file?

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/main/java/org/kie/server/remote/rest/jbpm/ui/ImageResource.java [hunk 17]
```diff
@@ -104,14 +105,15 @@ public Response getProcessInstanceImage(@javax.ws.rs.core.Context HttpHeaders he
...
         Header conversationIdHeader = buildConversationIdHeader(containerId, context, headers);
         try {
             String svgString = imageServiceBase.getActiveProcessImage(containerId, procInstId,
                                                                       (COMPLETED_COLOR.equals(svgNodeCompletedColor) ? COMPLETED_COLOR : URLDecoder.decode(svgNodeCompletedColor, "UTF-8")),
                                                                       (COMPLETED_BORDER_COLOR.equals(svgNodeCompletedBorderColor) ? COMPLETED_BORDER_COLOR : URLDecoder.decode(svgNodeCompletedBorderColor, "UTF-8")),
-                                                                      (ACTIVE_BORDER_COLOR.equals(svgActiveNodeBorderColor) ? ACTIVE_BORDER_COLOR : URLDecoder.decode(svgActiveNodeBorderColor, "UTF-8")), showBadges);
+                                                                      (ACTIVE_BORDER_COLOR.equals(svgActiveNodeBorderColor) ? ACTIVE_BORDER_COLOR : URLDecoder.decode(svgActiveNodeBorderColor, "UTF-8")), showBadges, (ERROR_BORDER_COLOR.equals(svgErrorNodeBorderColor) ? ERROR_BORDER_COLOR : URLDecoder.decode(svgErrorNodeBorderColor, "UTF-8")));
```
Comment (jstastny-cz) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/main/java/org/kie/server/remote/rest/jbpm/ui/ImageResource.java:116]: Could we introduce a private method with two args instead of that repetitive ternary if?

Or at least move to a separate line.

**Review by jstastny-cz — CHANGES_REQUESTED** [review]

**Comment by bxf12315** [comment]
@jstastny-cz Done

**Comment by bxf12315** [comment]
Jenkins retest this

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json [hunk 18]
```diff
@@ -22,11 +22,42 @@
       }
     }
   },
-
   "ignores": {
     "revapi": {
-    "_comment": "Changes between 7.48.0.Final and the current branch. These changes are desired and thus ignored.",
-    "ignore": [
+      "_comment": "Changes between 7.63.0.Final and the current branch. These changes are desired and thus ignored.",
```
Comment (nmirasch) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json:27]: @bxf12315 it seems the reference version is 7.52.0.Final instead of 7.63.0.Final

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json [hunk 19]
```diff
@@ -22,11 +22,42 @@
...
+      "ignore": [
+        {
+          "code": "java.annotation.attributeValueChanged",
```
Comment (nmirasch) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json:30]: from line 30 --> 60 have to be removed regarding is duplicated at the end of "ignore" list

File: kie-server-parent/kie-server-remote/kie-server-client/src/build/revapi-config.json [hunk 20]
```diff
@@ -95,8 +93,20 @@
               "methodName": "suspendTask",
               "elementKind": "method",
               "justification": "[JBPM-10015] Allow to set suspendUntil during user task suspend operation as parameter."
+            },
+            {
```
Comment (nmirasch) [inline kie-server-parent/kie-server-remote/kie-server-client/src/build/revapi-config.json:97]: @bxf12315 this block is the only change is needed here (regarding numberOfParametersChanged)

**Review by nmirasch — CHANGES_REQUESTED** [review]

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json (hunk 19 above)
Comment (bxf12315) [reply kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json:30]: it's not duplicated, the second one is related to swagger annotation.

Comment (nmirasch) [reply kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json:30]: @bxf12315 what I mean is in this file we only need the three last blocks you added to ignore list. This first two ones are duplicated and have to be removed.

File: kie-server-parent/kie-server-remote/kie-server-client/src/build/revapi-config.json (hunk 16 above)
Comment (nmirasch) [reply kie-server-parent/kie-server-remote/kie-server-client/src/build/revapi-config.json:2]: +1

**Review by nmirasch — CHANGES_REQUESTED** [review]

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json (hunk 19 above)
Comment (bxf12315) [reply kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json:30]: them different 
1  "annotationType": "javax.ws.rs.DefaultValue",
          "annotation": "@javax.ws.rs.DefaultValue(\"#1e90ff\")",
2  "annotationType": "io.swagger.annotations.ApiParam",
          "annotation": "@io.swagger.annotations.ApiParam(value = \"svg active node border color\", required = false, example = \"#1e90ff\")",

Comment (nmirasch) [reply kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json:30]: @bxf12315 Line 50 is equals to L106 and L34 = L122

Comment (bxf12315) [reply kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/build/revapi-config.json]: Done

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java [hunk 21]
```diff
@@ -135,6 +136,10 @@ public String getActiveProcessImage(String containerId, long procInstId, String
             Collection<NodeInstanceDesc> activeLogs = dataService.getProcessInstanceHistoryActive(procInstId, qc);
             Collection<NodeInstanceDesc> completedLogs = dataService.getProcessInstanceHistoryCompleted(procInstId, qc);
             Collection<NodeInstanceDesc> fullLogs = dataService.getProcessInstanceFullHistory(procInstId, qc);
+
+            List<String> keyList = fullLogs.stream().collect(Collectors.groupingBy(NodeInstanceDesc::getNodeId, Collectors.counting())).entrySet().stream().filter(e -> e.getValue() == 1).map(Map.Entry::getKey).collect(Collectors.toList());
+            List<String> errors = fullLogs.stream().filter(nodeInstanceDesc -> (((org.jbpm.kie.services.impl.model.NodeInstanceDesc) nodeInstanceDesc).getType() == 6) && keyList.contains(nodeInstanceDesc.getNodeId())).map(NodeInstanceDesc::getNodeId).collect(Collectors.toList());
```
Comment (nmirasch) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java:141]: @bxf12315 Type = 6 is not corresponding with the nodes in error https://github.com/kiegroup/droolsjbpm-knowledge/blob/7.x/kie-api/src/main/java/org/kie/api/runtime/manager/audit/NodeInstanceLog.java#L64

**Review by nmirasch — CHANGES_REQUESTED** [review]

**Comment by bxf12315** [comment]
Jenkins retest this

**Comment by barboras7** [comment]
Hi @nmirasch @bxf12315,
I see some inconsistency between this PR and https://github.com/kiegroup/droolsjbpm-integration/pull/2732.
What is the agreed name for the async node with red border? In code of this PR, it is called "error node" and in the other PR, it is called "async active node" or "failing active node". Could you please agree on what name should be the correct one and fix the PRs accordingly?
The code freeze is today so please, if you could agree on something and propagate it through the PR, that would be great.

**Comment by bxf12315** [comment]
Jenkins retest this

**Comment by bxf12315** [comment]
Jenkins retest this

**Comment by nmirasch** [comment]
Jenkins retest this

**Comment by nmirasch** [comment]
@bxf12315   could you rename the commit to "JBPM-9979: Mark async active nodes and change default active node border color" to reflect what it is really adding/changing?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/test/java/org/kie/server/services/jbpm/ui/ImageServiceBaseTest.java [hunk 22]
```diff
@@ -201,6 +201,73 @@ public void testLoopSubProcess() throws Exception {
...
+
+        checkStrokeAttributeAtNode(svgDocument, humanTaskOne, ACTIVE_BORDER_COLOR);
+        checkStrokeAttributeAtNode(svgDocument, oneScriptNode, COMPLETED_BORDER_COLOR);
+        checkStrokeAttributeAtNode(svgDocument, twoScriptNode, COMPLETED_BORDER_COLOR);
+        checkStrokeAttributeAtNode(svgDocument, threeScriptNode, ACTIVE_ASYNC_BORDER_COLOR);
+
+        String completedNodeColor = "black";
+        String completedNodeBorderColor = "grey";
+        String activeNodeBorderColor = "blue";
+        String errorNodeBorderColor = "red";
```
Comment (barboras7) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/test/java/org/kie/server/services/jbpm/ui/ImageServiceBaseTest.java:259]:
```suggestion
        String activeAsyncNodeBorderColor = "red";
```

**Review by barboras7 — CHANGES_REQUESTED** [review]

**Comment by barboras7** [comment]
Jenkins retest this

**Comment by barboras7** [comment]
jenkins retest this

**Review by barboras7 — APPROVED** [review]

**Comment by nmirasch** [comment]
jenkins retest this

**Comment by nmirasch** [comment]
jenkins retest this

**Comment by nmirasch** [comment]
jenkins retest this

**Comment by bxf12315** [comment]
jenkins retest this

**Comment by jstastny-cz** [comment]
Please update the description (first comment) to point at https://issues.redhat.com/browse/JBPM-9979.

**Comment by bxf12315** [comment]
> Please update the description (first comment) to point at https://issues.redhat.com/browse/JBPM-9979.

Done

**Comment by nmirasch** [comment]
jenkins retest this

**Review by nmirasch — APPROVED** [review]

**Comment by bxf12315** [comment]
jenkins retest this

**Comment by nmirasch** [comment]
Jenkins retest this
